pkgs/tools/misc/powerline-rs/default.nix

```diff
@@ -1,28 +1,25 @@
-{ stdenv, lib, rustPlatform, fetchFromGitLab, pkgconfig, file, perl, curl, cmake, openssl, libssh2, libgit2, libzip, Security }:
-rustPlatform.buildRustPackage rec {
+{ callPackage, lib, fetchFromGitLab }:
+
+let
   pname = "powerline-rs";
-  version = "0.1.9";
+  version = "0.2.0";
 
-  src = fetchFromGitLab {
+  package = callPackage (fetchFromGitLab {
     owner = "jD91mZM2";
-    repo = "powerline-rs";
-    #rev = version;
-
-    # Support for $COMPLETION_OUT:
-    rev = "44679385a95dd9f3ebd9b093f9ef8925610e9a23";
-
-    sha256 = "1mxkw6ydnqjyplbki2j9pbnlhxmkw9qqw54443a3cjmn2g08jyzp";
-  };
-
-  cargoSha256 = "1d0f1c1vp1r9r3ic921xkcr59f4a45y2xbxm4gl6grhb9z6p5k7l";
+    repo = pname;
+    # Patch for default.nix to basically just call `.build`, see the
+    # commit for more details.
+    rev = "e4fcd19659e7f5ff56ffc6a57a0fdbc0280dc5bd";
+    # rev = version;
 
-  nativeBuildInputs = [ pkgconfig file perl cmake curl ];
-  buildInputs = [ openssl libssh2 libgit2 libzip ] ++ lib.optional stdenv.isDarwin Security;
+    sha256 = "0a1frfn8jpbbrk9xg305pgwzl3dnk6g95ar8fpsrk4sly2alhsj2";
+  }) {};
```

Comment:
you call your default.nix with only callPackage, lib and FetchFromGitLab. Yet, it expects a "pkgs"

Honestly, i would just use the normal rustPlatform.buildRustPackage workflow that you had before,

Reply:
callPackage should fill in all the necessary arguments